packages/web/src/pages/search-page-v2/SearchFilters.tsx

```diff
@@ -106,6 +107,80 @@ const MoodFilter = () => {
   )
 }
 
+const keyArr = [
+  'C',
+  'C#/Db',
+  'D',
+  'D#/Eb',
+  'E',
+  'F',
+  'F#/Gb',
+  'G',
+  'G#/Ab',
+  'A',
+  'A#/Bb',
+  'B'
+]
+
+const KeyFilter = () => {
+  const [urlSearchParams] = useSearchParams()
+  const key = urlSearchParams.get('key')
+  const updateSearchParams = useUpdateSearchParams('key')
+  const [scale, setScale] = useState<'Major' | 'Minor'>('Major')
+  const keyOptions = keyArr.map((key) => ({
+    label: key,
+    value: key
+  }))
+
+  return (
+    <FilterButton
+      value={key}
+      label={key ?? messages.key}
+      onChange={updateSearchParams}
+      iconRight={IconCaretDown}
+    >
+      {({ handleChange, isOpen, setIsOpen, anchorRef }) => (
```

Comment:
maybe a little off that we have to implement the Popup part of this? my expectation is that filter button handles everything around allowing you to add a section before the options?

Reply:
yea you might be right. This is how Seb made the update to allow passing in the children so the options wont always be there, but the popup seems like it should be handled by the base component.

Reply:
looked at it for a bit and the default filtering functionality makes this a little difficult to do, but ill make a note to try to fix this up in a nicer way